src/libYARP_dev/src/devices/RGBDSensorWrapper/RGBDSensorWrapper.cpp

```diff
@@ -194,18 +195,20 @@ bool RGBDSensorWrapper::open(yarp::os::Searchable &config)
 
      setId("RGBDSensorWrapper for " + depthFrame_StreamingPort_Name);
 
-    if(!initialize_YARP(config) )
+    if(use_YARP && !initialize_YARP(config))
     {
         yError() << sensorId << "Error initializing YARP ports";
         return false;
     }
 
-    if(!initialize_ROS(config) )
+    if(use_ROS && !initialize_ROS(config))
     {
         yError() << sensorId << "Error initializing ROS topic";
         return false;
     }
-    return true;
+
+    RateThread::setRate(rate);
+    return RateThread::start();
```

Comment:
NO!! This is wrong!
In case of  'deferred attach' the thread wil start too early,  when the actual device does not exist yet.

The start has to be placed back where it was.
The correct sequence shall be 
- read the config file
- initialize yarp
- attach subdevice
- start thread.

So the openSubDeice has to be extracted from the 'fromConfig' and placed after the initializeYarp/Ros